core/font.lua

```diff
@@ -20,19 +23,36 @@ SILE.registerCommand("font", function (options, content)
   if options.features then SILE.settings:set("font.features", options.features) end
   if options.direction then SILE.settings:set("font.direction", options.direction) end
   if options.language then
-    if options.language ~= "und" and icu and icu.canonicalize_language then
-      local newlang = icu.canonicalize_language(options.language)
-      -- if newlang ~= options.language then
-        -- SU.warn("Language '"..options.language.."' not canonical, '"..newlang.."' will be used instead.")
-      -- end
-      options.language = newlang
-    end
+    -- BEGIN OMIKHLEIA HACKLANG
+    -- Commented out. Too soon and or at discrepancy with the "language" command anyway.
+    -- So let document.language is what the user wants, and loadLanguage take care of it
+    --
+    -- if options.language ~= "und" and icu and icu.canonicalize_language then
+    --   local newlang = icu.canonicalize_language(options.language)
+    --   -- if newlang ~= options.language then
+    --     -- SU.warn("Language '"..options.language.."' not canonical, '"..newlang.."' will be used instead.")
+    --   -- end
+    --   options.language = newlang
+    -- end
+    -- END OMIKHLEIA HACKLANG
     SILE.settings:set("document.language", options.language)
-    fluent:set_locale(options.language)
+    -- BEGIN OMIKHLEIA HACKLANG
+    -- Commented out. This is BAD design:
```

Comment:
As you note Fluent-lua does maintain state: the current locale—and also the messages loaded in each locale. Since SILE can and does flip back and forth between calling things for different locales rather than just setting one and sticking to it, the only way the Fluent API should be used from SILE at this point is to *always always* set the locale before all message operations. That is being done everywhere else and should be done here too. It should never be used to read or write a value without explicitly setting a locale.

Reply:
(I don't like state-maintaining APIs, and think it could have been avoided.)

_Notwithstanding_, it doesn't have to be done _here_ (in `\font`), that's the point of this commenting out:
- `SILE.languageSupport.loadLanguage()` does it anyway _when it needs to load messages_.
- The `\fluent` command does it when invoked for _reading_ and rendering a message

Reply:
A longer term proper usage here would be to instantiate and embed the correct Fluent messages class inside languageSupport when a language is loaded. The only "state" we're actually making use of is a toggle switch for which locale to access, the right way is to have each SILE language support module have it's own instance with the correlated locale.

Reply:
This has been addressed since this PR and comments were created.